File: Miscellaneous_Permutation.java
package miscellaneous;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Miscellaneous {

    public static void main(String[] args) {
        List<String> goods = new ArrayList<>();
        goods.add("Eno");
        goods.add("Chidi");
        goods.add("Olu");
        goods.add("Ahmed");
        goods.add("Osas");
        goods.add("Gbeda");
        Permutation perm = new Permutation();
        List<String[]> result = perm.possibleWordPermutations(goods, 3);
        System.out.println(perm.words + " permutation " + perm.r + ":\n");
        for (String[] set : result) {
            System.out.println(Arrays.toString(set) + ";");
        }
        System.out.println("\nNumber of ways is " + result.size() + ".");
    }

}

File: Miscellaneous_Permutation.pl
#!/usr/bin/perl;

use strict;
use warnings;

use PERMUTATION;

my @subjects = ("Eno", "Chidi", "Olu", "Ahmed", "Osas", "Gbeda");
my $r = 4;

# Use the permutation module/class
my $perm = PERMUTATION->new();

# $result receives an array reference of references
my $result = $perm->possibleWordPermutations(\@subjects, $r);

print ("[", join(", ", @subjects), "] permutation ", $r, ":\n\n");
# for each array reference in a dereferenced '$result'
print (join(", ", @{$_}), "\n") for @{$result};
print ("\n\nNumber of ways is ", scalar @{$result}, ".");

print "\n\n";